src/main/java/com/couchbase/client/encryption/CachingKeyring.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2020 Couchbase, Inc.
+ *
+ * Use of this software is subject to the Couchbase Inc. Enterprise Subscription License Agreement v7
+ * which may be found at https://www.couchbase.com/ESLA01162020.
+ */
+
+package com.couchbase.client.encryption;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+
+import java.time.Duration;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Wraps another Keyring and caches the results of "get" calls.
+ * <p>
+ * Requires the optional {@code com.github.ben-manes.caffeine:caffeine}
```

Comment:
What do you think about the idea to repackage and bundle it?

Reply:
Having the hashicorp stuff optional makes sense to me since it's for that specific third party product. Here it is different since we utilize the library for transparent caching?